### PR TITLE
Add Implication and Remove Nesting

### PR DESCRIPTION
**Pull request by beckydvn**
-Adds >> functionality to Var
-Creates an option to auto-simplify formulas upon creation by changing `auto_simplify` in the init module to True

### CODE REVIEW

**Comment by haz** [comment]
This looks pretty good to me! Can you throw in some tests [here](https://github.com/QuMuLab/python-nnf/blob/master/test_nnf.py) to make sure things work (mainly useful for future regressions). Be sure to test multiple children in the left/right formula (e.g., `(x & y) & (w & z)`).

File: nnf/__init__.py [hunk 1]
```diff
@@ -83,19 +84,33 @@ def all_models(names: 't.Iterable[Name]') -> t.Iterator[Model]:
...
+        if auto_simplify:
+            if type(self) == And:
+                return And({*self.children, *other.children}) if type(other) == And else And({*self.children, other})
         return And({self, other})
 
     def __or__(self: T_NNF, other: U_NNF) -> 'Or[t.Union[T_NNF, U_NNF]]':
         """Or({self, other})"""
+        # prevent unnecessary nesting
+        if auto_simplify:
+            if type(self) == Or:
```
Comment (blyxxyz) [inline nnf/__init__.py:106]: For some reason Mypy doesn't get it if you write it like this, this way should work:
```suggestion
            if isinstance(self, Or):
```
(that goes for all the other places with `type(...) == ...` too)

File: nnf/__init__.py [hunk 2]
```diff
@@ -83,19 +84,33 @@ def all_models(names: 't.Iterable[Name]') -> t.Iterator[Model]:
...
+                return Or({*self.children, *other.children}) if type(other) == Or else Or({*self.children, other})
         return Or({self, other})
 
+    def __rshift__(self: T_NNF, other: U_NNF)-> 'Or[t.Union[T_NNF.negate(), U_NNF]]':
```
Comment (blyxxyz) [inline nnf/__init__.py:110]: Mypy doesn't understand this. I think I went a little overboard with the `T_NNF` and `U_NNF` in the other signatures, it makes things complicated for only minor benefit.

This works:
```suggestion
    def __rshift__(self, other: 'NNF')-> 'Or[NNF]':
```

**Review by blyxxyz — CHANGES_REQUESTED** [review]
Tox/flake8 has some complaints about formatting, can you try to fix those? [`CONTRIBUTING.md`](https://github.com/QuMuLab/python-nnf/blob/master/CONTRIBUTING.md) has instructions for running it but let me know if it's unclear.

After you fix the formatting it might complain about Mypy, I left tips for that in the review.

**Review by blyxxyz — APPROVED** [review]
Looks good. Thank you for bearing with me.

**Comment by beckydvn** [comment]
No worries, thanks for taking a look! Hope it proves to be useful in the long run :)

**Review by haz — APPROVED** [review]

**Pull request merged**